File: Backend/TrackrAPI/TrackrAPI/Services/GestionEntidad/EntidadEstructuraValorService.cs
using TrackrAPI.Dtos.GestionEntidad;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.GestionEntidad;
using System.Collections.Generic;
using System.Linq;

namespace TrackrAPI.Services.GestionEntidad
{
    public class EntidadEstructuraValorService
    {
        private readonly IEntidadEstructuraValorRepository entidadEstructuraValorRepository;

        public EntidadEstructuraValorService(IEntidadEstructuraValorRepository entidadEstructuraValorRepository)
        {
            this.entidadEstructuraValorRepository = entidadEstructuraValorRepository;
        }

        public void Guardar(List<EntidadEstructuraValor> entidadEstructuraValorList)
        {
            var valoresEliminar = entidadEstructuraValorList.Where(campoValor =>
                    (campoValor.IdEntidadEstructuraValor > 0 && string.IsNullOrWhiteSpace(campoValor.Valor))
                || (campoValor.IdEntidadEstructuraValor > 0 && campoValor.Valor == "false"));

            var valoresAgregar = entidadEstructuraValorList.Where(campoValor =>
                !string.IsNullOrWhiteSpace(campoValor.Valor) && campoValor.IdEntidadEstructuraValor == 0);

            var valoresEditar = entidadEstructuraValorList.Where(campoValor =>
                !string.IsNullOrWhiteSpace(campoValor.Valor) && campoValor.IdEntidadEstructuraValor > 0);

            entidadEstructuraValorRepository.Agregar(valoresAgregar);
            entidadEstructuraValorRepository.Editar(valoresEditar);
            entidadEstructuraValorRepository.Eliminar(valoresEliminar);
        }

        public IEnumerable<EntidadEstructuraValorDto> ConsultarPorTabulacion(int idEntidadEstructura, int idTabla)
        {
            var valores = entidadEstructuraValorRepository.ConsultarPorTabulacion(idEntidadEstructura, idTabla);
            return valores;
        }
    }
}

File: Frontend/trackr-administrador/src/app/shared/http/gestion-entidad/entidad-estructura-valor.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { EntidadEstructuraValor } from '@models/gestion-entidad/entidad-estructura-valor';

@Injectable()
export class EntidadEstructuraValorService {
  private dataUrl = 'entidadEstructuraValor/';

  constructor(public http: HttpClient) {}

  consultarPorTabulacion(idEntidadEstructura: number, idTabla: number) {
    return this.http.get<EntidadEstructuraValor[]>(this.dataUrl + `consultarPorTabulacion/${idEntidadEstructura},${idTabla}`);
  }

  guardar(entidadEstructuraValorList: any[]) {
    return this.http.post<any[]>(this.dataUrl + `guardar`, entidadEstructuraValorList);
  }

}
